### PR TITLE
Add ticket pricing window validation

### LINKED ISSUE #676: Ticket pricing and ticket type  `start_date` and `end_date` cannot be the same
Add validation to block start and end dates from being the same

### PR DESCRIPTION
**Pull request by Timo614**
Closes #676

This branch just adds validations to the ticket type pricing validation logic that confirm that a given ticket pricing start_date and end_date fall within the ticket type window.

Adjusted the validation unit test to take these two cases into account and added a postman integration test for adding with validation errors.

### CODE REVIEW

**Comment by Timo614** [comment]
Will need to be rebased once the FMT changes come in. Currently erroring out due to the massive number of adjustments the latest FMT logic has.

File: db/tests/unit/ticket_types.rs [hunk 1]
```diff
@@ -2,8 +2,6 @@ use bigneon_db::dev::TestProject;
 use bigneon_db::prelude::*;
 use bigneon_db::utils::errors::ErrorCode::ValidationError;
 use chrono::NaiveDate;
-use diesel::result::Error;
-use diesel::Connection;
 use uuid::Uuid;
```
Comment (Krakaw) [inline db/tests/unit/ticket_types.rs]: Possibly adding this test case 
```suggestion
#[test]
pub fn create_with_same_date_validation_errors() {
    let project = TestProject::new();
    let connection = project.get_connection();
    let event = project.create_event().finish();
    let wallet_id = event.issuer_wallet(connection).unwrap().id;
    let same_date = NaiveDate::from_ymd(2016, 7, 8).and_hms(4, 10, 11);
    let result = event.add_ticket_type(
        "VIP".to_string(),
        None,
        100,
        same_date,
        same_date,
        wallet_id,
        None,
        0,
        connection,
    );
    match result {
        Ok(_) => {
            panic!("Expected validation error");
        }
        Err(error) => match &error.error_code {
            ValidationError { errors } => {
                assert!(errors.contains_key("start_date"));
                assert_eq!(errors["start_date"].len(), 1);
                assert_eq!(
                    errors["start_date"][0].code,
                    "start_date_must_be_before_end_date"
                );
                assert_eq!(
                    &errors["start_date"][0]
                        .message
                        .clone()
                        .unwrap()
                        .into_owned(),
                    "Start date must be before end date"
                );
            }
            _ => panic!("Expected validation error"),
        },
    }
}
```

**Comment by Krakaw** [comment]
Should we also add this to `db/tests/unit/ticket_pricing.rs`
```rust
#[test]
fn create_with_same_date_validation_errors() {
    let project = TestProject::new();
    let event = project.create_event().with_tickets().finish();
    let ticket_type = &event.ticket_types(project.get_connection()).unwrap()[0];
    let same_date = NaiveDate::from_ymd(2016, 7, 9).and_hms(4, 10, 11);

    let mut ticket_pricing = TicketPricing::create(
        ticket_type.id,
        "Early Bird".to_string(),
        same_date,
        same_date,
        100,
        false,
    );

    let result = ticket_pricing.clone().commit(project.get_connection());
    match result {
        Ok(_) => {
            panic!("Expected validation error");
        }
        Err(error) => match &error.error_code {
            ValidationError { errors } => {
                assert!(errors.contains_key("ticket_pricing.start_date"));
                assert_eq!(errors["ticket_pricing.start_date"].len(), 1);
                assert_eq!(
                    errors["ticket_pricing.start_date"][0].code,
                    "start_date_must_be_before_end_date"
                );
                assert_eq!(
                    &errors["ticket_pricing.start_date"][0]
                        .message
                        .clone()
                        .unwrap()
                        .into_owned(),
                    "Start date must be before end date"
                );
            }
            _ => panic!("Expected validation error"),
        },
    }

    // Period without start date validation
    ticket_pricing.start_date = same_date;
    ticket_pricing.end_date = NaiveDate::from_ymd(2016, 7, 15).and_hms(4, 10, 11);
    let result = ticket_pricing.clone().commit(project.get_connection());
    assert!(result.is_ok());
}
```

**Comment by Timo614** [comment]
Added those two test cases. Travis seems to have a major backlog to work through at the moment so will hold off on updating and merging the branch until that all settles.

**Pull request merged**